tests/client_test.go

```diff
@@ -955,3 +969,59 @@ func compareStreamUpstreamServers(x []client.StreamUpstreamServer, y []client.St
 
 	return reflect.DeepEqual(xServers, yServers)
 }
+
+func TestUpstreamServerWithDrain(t *testing.T) {
+	httpClient := &http.Client{}
+	c, err := client.NewNginxClient(httpClient, "http://127.0.0.1:8080/api")
+	if err != nil {
+		t.Fatalf("Error connecting to nginx: %v", err)
+	}
+
+	maxFails := 64
+	weight := 10
+	server := client.UpstreamServer{
```

Comment:
This test case is just to test drain. There is a lot of unrelated data in this struct. 

Instead this struct should match the structs from the `Default` test cases.

Something like:
```go
	server := client.UpstreamServer{
		Server:      "127.0.0.1:2000",
		MaxFails:    &defaultMaxFails,
		FailTimeout: "10s",
		SlowStart:   "0s",
		Weight:      &defaultWeight,
	}
```